src/main/connect/connectionManager.ts

```diff
@@ -257,47 +374,108 @@ export class ConnectionManager implements ExtensionComponent {
         await this._context.globalState.update(ConnectionManager.RT_URL_KEY, this._rtUrl);
     }
 
-    private async setUsername(prompt: boolean): Promise<boolean> {
-        if (prompt) {
-            this._username =
-                (await vscode.window.showInputBox({
-                    prompt: 'Enter username',
-                    value: this._username,
-                    ignoreFocusOut: true,
-                    validateInput: ConnectionUtils.validateFieldNotEmpty
-                })) || '';
-        } else {
-            this._username = (await this._context.globalState.get(ConnectionManager.XRAY_USERNAME_KEY)) || '';
-        }
+    private async setUsernameFromFilesystem(): Promise<boolean> {
+        this._username = (await this._context.globalState.get(ConnectionManager.XRAY_USERNAME_KEY)) || '';
+        return !!this._username;
+    }
+
+    private async promptUsername(): Promise<boolean> {
+        this._username =
+            (await vscode.window.showInputBox({
+                prompt: 'Enter username',
+                value: this._username,
+                ignoreFocusOut: true,
+                validateInput: ConnectionUtils.validateFieldNotEmpty
+            })) || '';
         return !!this._username;
     }
 
     private async storeUsername() {
         await this._context.globalState.update(ConnectionManager.XRAY_USERNAME_KEY, this._username);
     }
 
-    private async setPassword(prompt: boolean): Promise<boolean> {
-        if (prompt) {
-            this._password =
-                (await vscode.window.showInputBox({
-                    prompt: 'Enter password',
-                    password: true,
-                    ignoreFocusOut: true,
-                    validateInput: ConnectionUtils.validateFieldNotEmpty
-                })) || '';
-        } else {
-            this._password =
-                (await keytar.getPassword(ConnectionManager.SERVICE_ID, this.createAccountId(this._url, this._username))) ||
-                (await keytar.getPassword(ConnectionManager.SERVICE_ID, this.createAccountId(this._xrayUrl, this._username))) ||
-                '';
+    private async setPasswordFromFilesystem(): Promise<boolean> {
```

Comment:
I think we should make it clear that the password is not getting readen from a file.
```suggestion
    private async setPasswordFromKeyStore(): Promise<boolean> {
```